File: CS3305/lab00/Stattest.java
import java.util.Scanner;

public class Stattest {

    public static Scanner si = new Scanner( System.in ); //tie Scanner to keyboard

    public static void main( String[] args ) {
        Statistician s1, s2, s3;
        s1 = new Statistician(); s2 = new Statistician(); s3 = new Statistician();
        char choice = ' ';
        double x = 0.0;

        System.out.printf("\033c"); //clear screen
        do {
            System.out.println();
            print_menu( );
            choice = toupper( get_user_command( ) );
            switch (choice) {
                case 'R'  :  System.out.printf("\tWhich one should I reset ( 1, 2, 3 )?\n");
                             choice = get_user_command( );
                             switch (choice) {
                                 case '1' : s1.reset( ); break;
                                 case '2' : s2.reset( ); break;
                                 case '3' : s3.reset( ); break;
                                 default  : System.out.printf("\tTry again -- bad choice\n");
                                            //System.out.printf("\033c"); //clear screen
                                            continue;
                              }
                              System.out.printf("\tReset activated for s" + choice + "\n");
                              break;
                case '1'   :  s1.nextNumber( get_number( ) ); break;
                case '2'   :  s2.nextNumber( get_number( ) ); break;
                case '3'   :  s3.nextNumber( get_number( ) ); break;
                case 'D'   :  System.out.printf("\ts1 =  %s\n", s1);
                              System.out.printf("\ts2 =  %s\n", s2);
                              System.out.printf("\ts3 =  %s\n", s3);
                              break;
                case 'E'   :  if ( s1.equals(s2) )
                                  System.out.printf("\ts1 and s2 are equal\n");
                              else
                                  System.out.printf("\ts1 and s2 are not equal\n");
                              break;
                case '+'   :  s3 = Statistician.union(s1, s2);
                              System.out.printf("\ts3 has been set to union(s1, s2)\n");
                              break;
                case '*'   :  System.out.printf("\tPlease enter a value for x: ");
                              x = si.nextDouble();
                              s3 = Statistician.scale( x, s1 );
                              System.out.printf("\ts3 has been set to scale( x, s1 )\n");
                              break;
                case 'Q'   :  System.out.printf("\tRidicule is the best test of truth . . . \n");
                              break;
                default    :  System.out.printf("\t%s is invalid . . .  sorry, try again\n", choice);
                              //System.out.printf("\033c"); //clear screen
            } 
        } while ( ( choice != 'Q' ) );
    }


    public static void print_menu( ) {
        System.out.printf("\n");
        System.out.printf("\tThe following choices are available: \n");
        System.out.printf("\t R  Activate one of the reset( ) functions\n");
        System.out.printf("\t 1  Add a new number to the 1st Statistician s1\n");
        System.out.printf("\t 2  Add a new number to the 2nd Statistician s2\n");
        System.out.printf("\t 3  Add a new number to the 3rd Statistician s3\n");
        System.out.printf("\t D  Display the state of the statisticians s1, s2 and s3\n");
        System.out.printf("\t E  Test whether s1.equals(s2)\n");
        System.out.printf("\t +  Set Statistician s3 equal to union( s1, s2 )\n");
        System.out.printf("\t *  Set Statistician s3 equal to scale( x, s1 )\n");
        System.out.printf("\t Q  Quit this test program\n");
    }

    public static char get_user_command( ) {
        char command;

        System.out.printf("\tEnter choice: ");
        command = si.next( ).charAt(0);
        return command;
    }

    public static double get_number( ) {
        double result;

        System.out.printf("\tPlease enter the next real number for the sequence: ");
        result = si.nextDouble();
        return result;
    }

    public static char toupper( char aChar ) {
        if ( aChar >= 'a' && aChar <= 'z' ) {
            int temp = (int)aChar ^ 0x20;
            aChar = (char)temp;
        }
        return aChar;
    }

}

File: CS3305/lab01/stattest.cxx
// FILE: stattest.cxx
// An interactive test program for the statistician class

#include <cctype>    // Provides toupper
#include <iomanip>   // Provides setw to set the width of an output
#include <iostream>  // Provides cout, cin
#include <cstdlib>   // Provides EXIT_SUCCESS
#include "statistician.h"
using namespace std;

// PROTOTYPES of functions used in this test program:
void print_menu( );
// Postcondition: A menu of choices for this program has been written to cout.
// Library facilties used: iostream.h

char get_user_command( );
// Postcondition: The user has been prompted to enter a one character command.
// A line of input (with at least one character) has been read, and the first
// character of the input line is returned.

double get_number( );
// Postcondition: The user has been prompted to enter a real number. The
// number has been read, echoed to the screen, and returned by the function.

int main( ) {
    statistician s1, s2, s3;  // Three statisticians for us to play with
    char choice;              // A command character entered by the user
    double x;                 // Value for multiplication x*s1

    cout << "Three statisticians s1, s2, and s3 are ready to test." << endl;

    do {
        cout << endl;
        print_menu( );
        choice = toupper(get_user_command( ));
        switch (choice) {
            case 'R': cout << "Which one should I reset (1, 2, 3) " << endl;
                      choice = get_user_command( );
                      switch (choice) {
			  case '1': s1.reset( );
                                    break;
			  case '2': s2.reset( );
                                    break;
			  case '3': s3.reset( );
                                    break;
                          default : cout << "Try again -- bad choice\n";
                                    continue;
                      }
                      cout << "Reset activated for s" << choice << "." << endl;
                      break;
            case '1': s1.next(get_number( ));
                      break;
            case '2': s2.next(get_number( ));
                      break;
            case '3': s3.next(get_number( ));
                      break;
            case 'D': cout << "s1 = " << s1 << endl;
                      cout << "s2 = " << s2 << endl;
                      cout << "s3 = " << s3 << endl;
                      break;
            case 'E': if (s1 == s2)
                          cout << "s1 and s2 are equal." << endl;
                      else
                          cout << "s1 and s2 are not equal." << endl;
                      break;
            case '+': s3 = s1 + s2;
                      cout << "s3 has been set to s1 + s2" << endl;
                      break;
            case '*': cout << "Please type a value for x: ";
                      cin >> x;
                      s3 = x * s1;
                      cout << "s3 has been set to " << x << " * s1" << endl;
		      break;
            case 'Q': cout << "Ridicule is the best test of truth." << endl;
                      break;
            default: cout << choice << " is invalid. Sorry." << endl;
        }
    }
    while ((choice != 'Q'));

    return EXIT_SUCCESS;

}

void print_menu( ) {
    cout << endl;
    cout << "The following choices are available: " << endl;
    cout << " R  Activate one of the reset( ) functions" << endl;
    cout << " 1  Add a new number to the 1st statistician s1" << endl;
    cout << " 2  Add a new number to the 2nd statistician s2" << endl;
    cout << " 3  Add a new number to the 3rd statistician s3" << endl;
    cout << " D  Display the state of the statistians s1, s2 and s3" << endl;
    cout << " E  Test whether s1 == s2" << endl;
    cout << " +  Set the third statistician s3 equal to s1 + s2" << endl;
    cout << " *  Set the third statistician s3 equal to x*s1" << endl;
    cout << " Q  Quit this test program" << endl;
}

char get_user_command( ) {
// Library facilties used: iostream
    char command;

    cout << "Enter choice: ";
    cin >> command; 
   
    return command;
}

double get_number( ) {
// Library facilties used: iostream
    double result;

    cout << "Please enter the next real number for the sequence: ";
    cin  >> result;
    cout << result << " has been read." << endl;
    return result;
}

